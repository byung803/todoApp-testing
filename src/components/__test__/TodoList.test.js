import React from 'react';
import { shallow } from 'enzyme';
import TodoList from '../TodoList';
import { todosFixture } from './fixtures';

let wrapper;


describe('TodoList compo with empty todos', () => {

    beforeEach(() => {
        wrapper = shallow(<TodoList todos={[]} />);
    });

    it('should have a div', () => {
        expect(wrapper.find('div').length).toBe(1);
    })

    it('should return nothing when got empty todo list', () => {
        expect(wrapper).toMatchSnapshot();
    })
});

describe('TodoList compo with fixture todos', () => {

    beforeEach(() => {
        wrapper = shallow(<TodoList todos={todosFixture} />);
    });

    it('should return Todo Components when got todosFixture todo list', () => {
        expect(wrapper).toMatchSnapshot();
    })

});