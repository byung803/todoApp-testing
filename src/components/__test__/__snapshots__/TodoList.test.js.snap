// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList compo with empty todos should return nothing when got empty todo list 1`] = `<div />`;

exports[`TodoList compo with fixture todos should return Todo Components when got todosFixture todo list 1`] = `
<div>
  <Todo
    completed={false}
    key="1"
    removeTodo={[Function]}
    text="first"
    toggleTodo={[Function]}
  />
  <Todo
    completed={true}
    key="2"
    removeTodo={[Function]}
    text="second"
    toggleTodo={[Function]}
  />
  <Todo
    completed={false}
    key="3"
    removeTodo={[Function]}
    text="third"
    toggleTodo={[Function]}
  />
</div>
`;
